pkg/cri/sbserver/sandbox_run.go

```diff
@@ -212,6 +213,50 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		return nil, fmt.Errorf("unable to save sandbox %q to store: %w", id, err)
 	}
 
+	// Create sandbox container root directories.
```

Comment:
Can you explain a little more why this functionality was moved out of the podsandbox controller back to here? This logic should be implementation specific rather than done by by the CRI server.

Reply:
It seems the shim sandbox controller also need to implement  a same logic to create these files, otherwise the hostname, hosts and DNS configs is lost in containers. I didn't start a container with runc and sandbox_mode of shim successfully, But I think if we can succeed start it, these files should be created.

If we move it into podsandbox controller, I think we should copy these codes into the shim controller too. So I think maybe we should reserve it as a common logic in the CRI server.

Yes I think it is a implementation specific logic because some sandbox like wasm may not need these files, but I think most of the sandbox need these files.

I can remove this commit if it is neccesary, and implement it in our sandbox controller.

Reply:
It could make sense to break this functionality into a library which different implementations can use. We are trying to minimize the amount of host setup needed when using the sandbox interface. The sandbox implementations should own this rather than it being a common operation for any sandbox.

Reply:
OK，the commit is removed